frontend/plan/src/components/modals/NameScheduleModalInterior.js

```diff
@@ -1,13 +1,24 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import { validateScheduleName } from "../schedule/schedule_name_validation";
 
 const NameScheduleModalInterior = ({
-    usedScheduleNames, namingFunction, close, buttonName,
+    usedScheduleNames, namingFunction, close, buttonName, defaultValue, overwriteDefault = false,
 }) => {
     const [inputRef, setInputRef] = useState(null);
-    const [userInput, setUserInput] = useState("");
+    const [userInput, setUserInput] = useState(defaultValue);
     const { error, message: errorMessage } = validateScheduleName(userInput, usedScheduleNames);
+    useEffect(() => {
+        const listener = (event) => {
+            if (!userInput && inputRef && !inputRef.contains(event.target)) {
+                setUserInput(defaultValue);
+            }
+        };
+        document.addEventListener("click", listener);
+        return () => {
+            document.removeEventListener("click", listener);
+        };
+    });
```

Comment:
refactor to use Ying's OutClickable hook

Reply:
Ying's hook doesn't work here because I have to use the ref for something else too.